### PR TITLE
Support integer vid.

### PR DESCRIPTION
**Pull request by CPWstatic**

### CODE REVIEW

**Review by dutor — COMMENTED** [review]
The type of vertex id in schema-related contexts are `FIXED_STRING(INTEGER)`

File: src/service/Session.h [hunk 1]
```diff
@@ -25,17 +34,12 @@ class Session final {
...
     }
 
-    void setSpace(const std::string &name, GraphSpaceID space) {
-        spaceName_ = name;
-        space_ = space;
-    }
-
-    const std::string& spaceName() const {
-        return spaceName_;
+    void setSpace(SpaceInfo space) {
```
Comment (laura-ding) [inline src/service/Session.h:41]: it's better to structure the SpaceInfo in here.

File: src/service/Session.h [hunk 2]
```diff
@@ -79,9 +83,8 @@ class Session final {
 
 
 private:
-    int64_t           id_{0};
-    GraphSpaceID      space_{-1};
-    std::string       spaceName_;
+    int64_t           id_{kInvalidSpaceID};
```
Comment (laura-ding) [inline src/service/Session.h:86]: the `id_` is no need.

File: src/validator/AdminValidator.cpp [hunk 3]
```diff
@@ -8,83 +8,97 @@
...
+                       << meta::cpp2::_PropertyType_VALUES_TO_NAMES.at(typeDef.type);
+                    return Status::Error(ss.str());
+                }
+                spaceDesc_.vid_type = typeDef.type;
+
+                if (typeDef.type == meta::cpp2::PropertyType::INT64) {
+                    spaceDesc_.vid_size = 8;
+                } else {
+                    if (typeDef.typeLen <= 0 ||
+                        typeDef.typeLen > std::numeric_limits<int16_t>::max()) {
```
Comment (laura-ding) [inline src/validator/AdminValidator.cpp:59]: The thrift interface defines the typeLen is int32, why here limit int16?

File: src/validator/AdminValidator.cpp [hunk 4]
```diff
@@ -8,83 +8,97 @@
...
+                        return Status::Error("vid size is out of size: %d", typeDef.typeLen);
```
Comment (laura-ding) [inline src/validator/AdminValidator.cpp:60]: The first letter needs to capital

File: src/parser/ColumnTypeDef.h [hunk 5]
```diff
@@ -13,8 +13,36 @@ namespace nebula {
 struct ColumnTypeDef {
     explicit ColumnTypeDef(meta::cpp2::PropertyType t, int16_t l = 0) : type(t), typeLen(l) {}
 
+    std::string toString() const {
+        switch (this->type) {
```
Comment (laura-ding) [inline src/parser/ColumnTypeDef.h:17]: You can use `SchemaUtil::typeToString` directly.

File: src/parser/parser.yy [hunk 6]
```diff
@@ -126,7 +126,7 @@ static constexpr size_t MAX_ABS_INTEGER = 9223372036854775808ULL;
 %token KW_EDGE KW_EDGES KW_STEPS KW_OVER KW_UPTO KW_REVERSELY KW_SPACE KW_DELETE KW_FIND KW_REBUILD
 %token KW_TAG KW_TAGS KW_UNION KW_INTERSECT KW_MINUS
 %token KW_NO KW_OVERWRITE KW_IN KW_DESCRIBE KW_DESC KW_SHOW KW_HOSTS KW_PART KW_PARTS KW_ADD
-%token KW_PARTITION_NUM KW_REPLICA_FACTOR KW_CHARSET KW_COLLATE KW_COLLATION KW_VID_SIZE
+%token KW_PARTITION_NUM KW_REPLICA_FACTOR KW_CHARSET KW_COLLATE KW_COLLATION KW_VID_SIZE KW_VID_TYPE
```
Comment (laura-ding) [inline src/parser/parser.yy:129]: `VID_SIZE` can delete

File: src/service/Session.h (hunk 2 above)
Comment (CPWstatic) [reply src/service/Session.h:86]: This is session id.

File: src/validator/AdminValidator.cpp (hunk 3 above)
Comment (CPWstatic) [reply src/validator/AdminValidator.cpp:59]: It was defined `int16` in `ColumnTypeDef`.

Comment (laura-ding) [reply src/validator/AdminValidator.cpp:59]: I am so sorry, it’s my fault, I define the wrong type.  the parser has checked the range. need to modify the type to int32_t.

Comment (CPWstatic) [reply src/validator/AdminValidator.cpp:59]: ok.

File: src/executor/admin/SpaceExecutor.cpp [hunk 7]
```diff
@@ -43,19 +43,21 @@ folly::Future<Status> DescSpaceExecutor::execute() {
                                     "Name",
                                     "Partition Number",
                                     "Replica Factor",
-                                    "Vid Size",
                                     "Charset",
-                                    "Collate"};
+                                    "Collate",
+                                    "Vid Type"};
```
Comment (Shylock-Hg) [inline src/executor/admin/SpaceExecutor.cpp:48]: Don't show Vid size?

File: src/context/Iterator.cpp [hunk 8]
```diff
@@ -257,7 +257,7 @@ Value GetNeighborsIter::getVertex() const {
 
     auto segment = currentSeg();
     auto vidVal = getColumn(nebula::kVid);
-    if (!vidVal.isStr()) {
+    if (!vidVal.isStr() && vidVal.isInt()) {
```
Comment (Shylock-Hg) [inline src/context/Iterator.cpp:260]: Better to wrap it a function, such as isVid.

File: src/validator/ASTValidator.cpp [hunk 9]
```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
```
Comment (laura-ding) [inline src/validator/ASTValidator.cpp:1]: it may be deleted, why add it again?

File: src/validator/AdminValidator.cpp (hunk 3 above)
Comment (laura-ding) [inline src/validator/AdminValidator.cpp:59]: the parser has checked it

File: src/executor/admin/SpaceExecutor.cpp (hunk 7 above)
Comment (CPWstatic) [reply src/executor/admin/SpaceExecutor.cpp:48]: fixed_string(size) or int.

File: src/util/ToJson.h [hunk 10]
```diff
@@ -22,8 +22,8 @@ struct Value;
 struct HostAddr;
 
 namespace meta {
-struct SpaceDesc;
 namespace cpp2 {
+struct SpaceDesc;
```
Comment (Shylock-Hg) [inline src/util/ToJson.h:26]: class SpaceDesc.

**Review by dutor — APPROVED** [review]
Awesome!

**Review by Shylock-Hg — APPROVED** [review]
LGTM

File: src/parser/ColumnTypeDef.h (hunk 5 above)
Comment (yixinglu) [reply src/parser/ColumnTypeDef.h:17]: I prefer to replace SchemaUtil::typeToString implementation with following codes.

**Pull request merged**